// n8n-nodes-qdrant/dist/credentials/QdrantApi.credentials.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.QdrantApi = void 0;
class QdrantApi {
    constructor() {
        this.name = 'qdrantRestApi';
        this.displayName = 'Qdrant API';
        this.icon = 'file:qdrant.svg';
        this.documentationUrl = 'https://qdrant.tech/documentation/cloud/authentication/';
        this.properties = [
            {
                displayName: 'API Key',
                name: 'apiKey',
                type: 'string',
                typeOptions: {
                    password: true,
                },
                default: '',
                required: false,
                description: 'Optional API key to use for authentication',
            },
            {
                displayName: 'REST URL',
                name: 'url',
                type: 'string',
                default: 'http://localhost:6333',
                required: true,
                description: 'The REST URL of the Qdrant instance to connect to',
            },
        ];
        this.authenticate = {
            type: 'generic',
            properties: {
                headers: {
                    'api-key': '={{$credentials.apiKey}}',
                },
            },
        };
        this.test = {
            request: {
                baseURL: '={{$credentials.url}}',
                method: 'GET',
            },
        };
    }
}
exports.QdrantApi = QdrantApi;
