// n8n-nodes-qdrant/dist/nodes/Qdrant/Qdrant.node.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Qdrant = void 0;
const descriptions_1 = require("./descriptions");
const utils_1 = require("./descriptions/utils");
const inputs = ["main"];
const outputs = ["main"];
class Qdrant {
    constructor() {
        this.description = {
            displayName: 'Qdrant',
            name: 'qdrant',
            icon: 'file:qdrant.svg',
            group: ['transform'],
            version: 1,
            subtitle: '={{$parameter["operation"]}}',
            description: 'Official n8n node to interface with Qdrant - https://qdrant.tech',
            defaults: {
                name: 'Qdrant',
            },
            usableAsTool: true,
            inputs,
            outputs,
            credentials: [
                {
                    name: 'qdrantRestApi',
                    required: true,
                },
            ],
            requestDefaults: {
                baseURL: '={{$credentials.url}}',
                headers: {
                    'Content-Type': 'application/json',
                },
            },
            properties: [
                {
                    displayName: 'Resource',
                    name: 'resource',
                    type: 'options',
                    noDataExpression: true,
                    options: [
                        {
                            name: 'Collection',
                            value: 'collection',
                        },
                        {
                            name: 'Payload',
                            value: 'payload',
                        },
                        {
                            name: 'Point',
                            value: 'point',
                        },
                        {
                            name: 'Search',
                            value: 'search',
                        },
                        {
                            name: 'Vector',
                            value: 'vector',
                        },
                    ],
                    default: 'collection',
                },
                {
                    displayName: 'Operation',
                    name: 'operation',
                    type: 'options',
                    noDataExpression: true,
                    displayOptions: {
                        show: {
                            resource: ['collection'],
                        },
                    },
                    options: [
                        descriptions_1.listCollectionsOperation,
                        descriptions_1.createCollectionOperation,
                        descriptions_1.updateCollectionOperation,
                        descriptions_1.getCollectionOperation,
                        descriptions_1.collectionExistsOperation,
                        descriptions_1.deleteCollectionOperation,
                    ],
                    default: '',
                },
                {
                    displayName: 'Operation',
                    name: 'operation',
                    type: 'options',
                    noDataExpression: true,
                    displayOptions: {
                        show: {
                            resource: ['point'],
                        },
                    },
                    options: [
                        descriptions_1.upsertPointsOperation,
                        descriptions_1.retrievePointOperation,
                        descriptions_1.retrievePointsOperation,
                        descriptions_1.deletePointsOperation,
                        descriptions_1.countPointsOperation,
                        descriptions_1.scrollPointsOperation,
                        descriptions_1.batchUpdatePointsOperation,
                    ],
                    default: '',
                },
                {
                    displayName: 'Operation',
                    name: 'operation',
                    type: 'options',
                    noDataExpression: true,
                    displayOptions: {
                        show: {
                            resource: ['vector'],
                        },
                    },
                    options: [descriptions_1.updateVectorsOperation, descriptions_1.deleteVectorsOperation],
                    default: '',
                },
                {
                    displayName: 'Operation',
                    name: 'operation',
                    type: 'options',
                    noDataExpression: true,
                    displayOptions: {
                        show: {
                            resource: ['payload'],
                        },
                    },
                    options: [
                        descriptions_1.setPayloadOperation,
                        descriptions_1.overwritePayloadOperation,
                        descriptions_1.deletePayloadOperation,
                        descriptions_1.clearPayloadOperation,
                        descriptions_1.payloadFacetsOperation,
                        descriptions_1.createPayloadIndexOperation,
                        descriptions_1.deletePayloadIndexOperation,
                    ],
                    default: '',
                },
                {
                    displayName: 'Operation',
                    name: 'operation',
                    type: 'options',
                    noDataExpression: true,
                    displayOptions: {
                        show: {
                            resource: ['search'],
                        },
                    },
                    options: [
                        descriptions_1.queryPointsOperation,
                        descriptions_1.queryBatchPointsOperation,
                        descriptions_1.queryPointsGroupsOperation,
                        descriptions_1.matrixPairsOperation,
                        descriptions_1.matrixOffsetsOperation,
                    ],
                    default: '',
                },
                ...descriptions_1.listCollectionsFields,
                ...descriptions_1.createCollectionFields,
                ...descriptions_1.updateCollectionFields,
                ...descriptions_1.getCollectionFields,
                ...descriptions_1.collectionExistsFields,
                ...descriptions_1.deleteCollectionFields,
                ...descriptions_1.upsertPointsFields,
                ...descriptions_1.retrievePointFields,
                ...descriptions_1.retrievePointsFields,
                ...descriptions_1.deletePointsFields,
                ...descriptions_1.countPointsFields,
                ...descriptions_1.scrollPointsFields,
                ...descriptions_1.batchUpdatePointsFields,
                ...descriptions_1.updateVectorsFields,
                ...descriptions_1.deleteVectorsFields,
                ...descriptions_1.setPayloadFields,
                ...descriptions_1.overwritePayloadFields,
                ...descriptions_1.deletePayloadFields,
                ...descriptions_1.clearPayloadFields,
                ...descriptions_1.payloadFacetsFields,
                ...descriptions_1.createPayloadIndexFields,
                ...descriptions_1.deletePayloadIndexFields,
                ...descriptions_1.queryPointsFields,
                ...descriptions_1.queryBatchPointsFields,
                ...descriptions_1.queryPointsGroupsFields,
                ...descriptions_1.matrixPairsFields,
                ...descriptions_1.matrixOffsetsFields,
            ],
        };
        this.methods = {
            listSearch: utils_1.listSearch,
        };
    }
}
exports.Qdrant = Qdrant;
