// n8n-nodes-qdrant/dist/nodes/Qdrant/descriptions/BatchUpdatePoints.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.batchUpdatePointsFields = exports.batchUpdatePointsOperation = void 0;
const Commons_1 = require("./Commons");
exports.batchUpdatePointsOperation = {
    name: 'Batch Update Points',
    value: 'batchUpdatePoints',
    action: 'Batch Update Points',
    description: 'Batch update points, including their respective vectors and payloads',
    routing: {
        request: {
            method: 'POST',
            url: '=/collections/{{$parameter.collectionName}}/points/batch',
            body: {
                operations: '={{JSON.parse($parameter.operations)}}',
            },
            qs: {
                wait: '={{$parameter.wait}}',
                ordering: '={{$parameter.ordering}}',
            },
        },
    },
};
exports.batchUpdatePointsFields = [
    (0, Commons_1.collectionNameField)('batchUpdatePoints'),
    {
        displayName: 'Operations',
        name: 'operations',
        description: 'List of operations to perform. Can include upsert, update_vectors, set_payload, delete_payload, delete_vectors, and clear_payload operations.',
        default: '[]',
        type: 'json',
        required: true,
        displayOptions: {
            show: {
                operation: ['batchUpdatePoints'],
            },
        },
    },
    (0, Commons_1.waitField)('batchUpdatePoints'),
    (0, Commons_1.orderingField)('batchUpdatePoints'),
];
