// n8n-nodes-qdrant/dist/nodes/Qdrant/descriptions/ClearPayload.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.clearPayloadFields = exports.clearPayloadOperation = void 0;
const Commons_1 = require("./Commons");
exports.clearPayloadOperation = {
    name: 'Clear Payload',
    value: 'clearPayload',
    action: 'Clear Payload',
    description: 'Removes the entire payload for specified points',
    routing: {
        request: {
            method: 'POST',
            url: '=/collections/{{$parameter.collectionName}}/points/payload/clear',
            body: {
                points: '={{JSON.parse($parameter.points)}}',
                filter: '={{JSON.parse($parameter.filter)}}',
                shard_key: '={{JSON.parse($parameter.shardKey)}}',
            },
            qs: {
                wait: '={{$parameter.wait}}',
                ordering: '={{$parameter.ordering}}',
            },
        },
    },
};
exports.clearPayloadFields = [
    (0, Commons_1.collectionNameField)('clearPayload'),
    (0, Commons_1.pointIdsField)('clearPayload'),
    (0, Commons_1.filterField)('clearPayload'),
    (0, Commons_1.shardKeyField)('clearPayload'),
    (0, Commons_1.waitField)('clearPayload'),
    (0, Commons_1.orderingField)('clearPayload'),
];
