// n8n-nodes-qdrant/dist/nodes/Qdrant/descriptions/CountPoints.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.countPointsFields = exports.countPointsOperation = void 0;
const Commons_1 = require("./Commons");
exports.countPointsOperation = {
    name: 'Count Points',
    value: 'countPoints',
    action: 'Count Points',
    description: 'Counts the number of points that match a specified filtering condition',
    routing: {
        request: {
            method: 'POST',
            url: '=/collections/{{$parameter.collectionName}}/points/count',
            body: {
                filter: '={{JSON.parse($parameter.filter)}}',
                exact: '={{$parameter.exact}}',
                shard_key: '={{JSON.parse($parameter.shardKey)}}',
            },
            qs: {
                timeout: '={{$parameter.timeout}}',
            },
        },
    },
};
exports.countPointsFields = [
    (0, Commons_1.collectionNameField)('countPoints'),
    (0, Commons_1.filterField)('countPoints'),
    {
        displayName: 'Exact',
        name: 'exact',
        description: 'Whether tocount exact number of points. If false, count approximate number of points faster. Approximate count might be unreliable during the indexing process',
        default: true,
        type: 'boolean',
        displayOptions: {
            show: {
                operation: ['countPoints'],
            },
        },
    },
    (0, Commons_1.shardKeyField)('countPoints'),
    (0, Commons_1.timeoutField)('countPoints'),
];
