// n8n-nodes-qdrant/dist/nodes/Qdrant/descriptions/CreatePayloadIndex.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createPayloadIndexFields = exports.createPayloadIndexOperation = void 0;
const Commons_1 = require("./Commons");
exports.createPayloadIndexOperation = {
    name: 'Create Payload Index',
    value: 'createPayloadIndex',
    action: 'Create Payload Index',
    description: 'Creates a payload index for a field in the specified collection',
    routing: {
        request: {
            method: 'PUT',
            url: '=/collections/{{$parameter.collectionName}}/index',
            body: {
                field_name: '={{$parameter.fieldName}}',
                field_schema: '={{JSON.parse($parameter.fieldSchema)}}',
            },
            qs: {
                wait: '={{$parameter.wait}}',
                ordering: '={{$parameter.ordering}}',
            },
        },
    },
};
exports.createPayloadIndexFields = [
    (0, Commons_1.collectionNameField)('createPayloadIndex'),
    {
        displayName: 'Field Name',
        name: 'fieldName',
        description: 'Name of the field to create index for',
        default: '',
        type: 'string',
        required: true,
        displayOptions: {
            show: {
                operation: ['createPayloadIndex'],
            },
        },
    },
    {
        displayName: 'Field Schema',
        name: 'fieldSchema',
        description: 'Schema of the field to create index for',
        default: '{}',
        type: 'json',
        required: true,
        displayOptions: {
            show: {
                operation: ['createPayloadIndex'],
            },
        },
    },
    (0, Commons_1.waitField)('createPayloadIndex'),
    (0, Commons_1.orderingField)('createPayloadIndex'),
];
