// n8n-nodes-qdrant/dist/nodes/Qdrant/descriptions/DeletePayloadIndex.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.deletePayloadIndexFields = exports.deletePayloadIndexOperation = void 0;
const Commons_1 = require("./Commons");
exports.deletePayloadIndexOperation = {
    name: 'Delete Payload Index',
    value: 'deletePayloadIndex',
    action: 'Delete Payload Index',
    description: 'Deletes a payload index for a field in the specified collection',
    routing: {
        request: {
            method: 'DELETE',
            url: '=/collections/{{$parameter.collectionName}}/index/{{$parameter.fieldName}}',
            qs: {
                wait: '={{$parameter.wait}}',
                ordering: '={{$parameter.ordering}}',
            },
        },
    },
};
exports.deletePayloadIndexFields = [
    (0, Commons_1.collectionNameField)('deletePayloadIndex'),
    {
        displayName: 'Field Name',
        name: 'fieldName',
        description: 'Name of the field where to delete the index',
        default: '',
        type: 'string',
        required: true,
        displayOptions: {
            show: {
                operation: ['deletePayloadIndex'],
            },
        },
    },
    (0, Commons_1.waitField)('deletePayloadIndex'),
    (0, Commons_1.orderingField)('deletePayloadIndex'),
];
