// n8n-nodes-qdrant/dist/nodes/Qdrant/descriptions/DeletePayload.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.deletePayloadFields = exports.deletePayloadOperation = void 0;
const Commons_1 = require("./Commons");
exports.deletePayloadOperation = {
    name: 'Delete Payload',
    value: 'deletePayload',
    action: 'Delete Payload',
    description: 'Deletes specified payload keys from points',
    routing: {
        request: {
            method: 'POST',
            url: '=/collections/{{$parameter.collectionName}}/points/payload/delete',
            body: {
                keys: '={{JSON.parse($parameter.keys)}}',
                points: '={{JSON.parse($parameter.points)}}',
                filter: '={{JSON.parse($parameter.filter)}}',
                shard_key: '={{JSON.parse($parameter.shardKey)}}',
            },
            qs: {
                wait: '={{$parameter.wait}}',
                ordering: '={{$parameter.ordering}}',
            },
        },
    },
};
exports.deletePayloadFields = [
    (0, Commons_1.collectionNameField)('deletePayload'),
    {
        displayName: 'Keys',
        name: 'keys',
        description: 'List of payload keys to remove from payload',
        default: '[]',
        type: 'json',
        required: true,
        displayOptions: {
            show: {
                operation: ['deletePayload'],
            },
        },
    },
    (0, Commons_1.pointIdsField)('deletePayload'),
    (0, Commons_1.filterField)('deletePayload'),
    (0, Commons_1.shardKeyField)('deletePayload'),
    (0, Commons_1.waitField)('deletePayload'),
    (0, Commons_1.orderingField)('deletePayload'),
];
