// n8n-nodes-qdrant/dist/nodes/Qdrant/descriptions/DeletePoints.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.deletePointsFields = exports.deletePointsOperation = void 0;
const Commons_1 = require("./Commons");
exports.deletePointsOperation = {
    name: 'Delete Points',
    value: 'deletePoints',
    action: 'Delete Points',
    description: 'Delete specified points from the collection',
    routing: {
        request: {
            method: 'POST',
            url: '=/collections/{{$parameter.collectionName}}/points/delete',
            body: {
                points: '={{JSON.parse($parameter.points)}}',
                filter: '={{JSON.parse($parameter.filter)}}',
                shard_key: '={{JSON.parse($parameter.shardKey)}}',
            },
            qs: {
                wait: '={{$parameter.wait}}',
                ordering: '={{$parameter.ordering}}',
            },
        },
    },
};
exports.deletePointsFields = [
    (0, Commons_1.collectionNameField)('deletePoints'),
    (0, Commons_1.pointIdsField)('deletePoints'),
    (0, Commons_1.filterField)('deletePoints'),
    (0, Commons_1.shardKeyField)('deletePoints'),
    (0, Commons_1.waitField)('deletePoints'),
    (0, Commons_1.orderingField)('deletePoints'),
];
