// n8n-nodes-qdrant/dist/nodes/Qdrant/descriptions/DeleteVectors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.deleteVectorsFields = exports.deleteVectorsOperation = void 0;
const Commons_1 = require("./Commons");
exports.deleteVectorsOperation = {
    name: 'Delete Vectors',
    value: 'deleteVectors',
    action: 'Delete Vectors',
    description: 'Delete specified vectors from points',
    routing: {
        request: {
            method: 'POST',
            url: '=/collections/{{$parameter.collectionName}}/points/vectors/delete',
            body: {
                points: '={{JSON.parse($parameter.points)}}',
                filter: '={{JSON.parse($parameter.filter)}}',
                vector: '={{JSON.parse($parameter.vector)}}',
                shard_key: '={{JSON.parse($parameter.shardKey)}}',
            },
            qs: {
                wait: '={{$parameter.wait}}',
                ordering: '={{$parameter.ordering}}',
            },
        },
    },
};
exports.deleteVectorsFields = [
    (0, Commons_1.collectionNameField)('deleteVectors'),
    {
        displayName: 'Vector',
        name: 'vector',
        description: 'List of vector names to delete',
        default: '[]',
        type: 'json',
        required: true,
        displayOptions: {
            show: {
                operation: ['deleteVectors'],
            },
        },
    },
    (0, Commons_1.pointIdsField)('deleteVectors'),
    (0, Commons_1.filterField)('deleteVectors'),
    (0, Commons_1.shardKeyField)('deleteVectors'),
    (0, Commons_1.waitField)('deleteVectors'),
    (0, Commons_1.orderingField)('deleteVectors'),
];
