// n8n-nodes-qdrant/dist/nodes/Qdrant/descriptions/DistanceMatrixOffsets.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.matrixOffsetsFields = exports.matrixOffsetsOperation = void 0;
const Commons_1 = require("./Commons");
exports.matrixOffsetsOperation = {
    name: 'Matrix Offsets',
    value: 'matrixOffsets',
    action: 'Matrix Offsets',
    description: 'Retrieve sparse matrix of pairwise distances between points sampled from the collection',
    routing: {
        request: {
            method: 'POST',
            url: '=/collections/{{$parameter.collectionName}}/points/search/matrix/offsets',
            body: {
                filter: '={{JSON.parse($parameter.filter)}}',
                sample: '={{$parameter.sample}}',
                limit: '={{$parameter.limit}}',
                using: '={{$parameter.using}}',
                shard_key: '={{JSON.parse($parameter.shardKey)}}',
            },
            qs: {
                consistency: '={{$parameter.consistency}}',
                timeout: '={{$parameter.timeout}}',
            },
        },
    },
};
exports.matrixOffsetsFields = [
    (0, Commons_1.collectionNameField)('matrixOffsets'),
    (0, Commons_1.filterField)('matrixOffsets'),
    {
        displayName: 'Sample',
        name: 'sample',
        description: 'How many points to select and search within',
        default: 10,
        type: 'number',
        displayOptions: {
            show: {
                operation: ['matrixOffsets'],
            },
        },
    },
    {
        displayName: 'Limit',
        name: 'limit',
        default: 50,
        type: 'number',
        typeOptions: {
            minValue: 1,
        },
        description: 'Max number of results to return',
        displayOptions: {
            show: {
                operation: ['matrixOffsets'],
            },
        },
    },
    {
        displayName: 'Using',
        name: 'using',
        description: 'Define which vector name to use for querying. If missing, the default vector is used.',
        default: '',
        type: 'string',
        displayOptions: {
            show: {
                operation: ['matrixOffsets'],
            },
        },
    },
    (0, Commons_1.shardKeyField)('matrixOffsets'),
    (0, Commons_1.consistencyField)('matrixOffsets'),
    (0, Commons_1.timeoutField)('matrixOffsets'),
];
