// n8n-nodes-qdrant/dist/nodes/Qdrant/descriptions/QueryBatchPoints.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.queryBatchPointsFields = exports.queryBatchPointsOperation = void 0;
const Commons_1 = require("./Commons");
exports.queryBatchPointsOperation = {
    name: 'Query Batch Points',
    value: 'queryBatchPoints',
    action: 'Query Batch Points',
    description: 'Query multiple points in batch using various search methods',
    routing: {
        request: {
            method: 'POST',
            url: '=/collections/{{$parameter.collectionName}}/points/query/batch',
            body: {
                searches: '={{JSON.parse($parameter.searches)}}',
            },
            qs: {
                consistency: '={{$parameter.consistency}}',
                timeout: '={{$parameter.timeout}}',
            },
        },
    },
};
exports.queryBatchPointsFields = [
    (0, Commons_1.collectionNameField)('queryBatchPoints'),
    {
        displayName: 'Searches',
        name: 'searches',
        description: 'List of search requests to perform in batch. Each request can be a vector search, ID search, or complex query.',
        default: '[]',
        type: 'json',
        required: true,
        displayOptions: {
            show: {
                operation: ['queryBatchPoints'],
            },
        },
    },
    (0, Commons_1.consistencyField)('queryBatchPoints'),
    (0, Commons_1.timeoutField)('queryBatchPoints'),
];
