// n8n-nodes-qdrant/dist/nodes/Qdrant/descriptions/ScrollPoints.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.scrollPointsFields = exports.scrollPointsOperation = void 0;
const Commons_1 = require("./Commons");
exports.scrollPointsOperation = {
    name: 'Scroll Points',
    value: 'scrollPoints',
    action: 'Scroll Points',
    description: 'Returns all points in a page-by-page manner',
    routing: {
        request: {
            method: 'POST',
            url: '=/collections/{{$parameter.collectionName}}/points/scroll',
            body: {
                offset: '={{isNaN(Number($parameter.offset)) ?  $parameter.offset : Number($parameter.offset)}}',
                limit: '={{$parameter.limit}}',
                filter: '={{JSON.parse($parameter.filter)}}',
                with_payload: '={{$parameter.withPayload}}',
                with_vector: '={{$parameter.withVector}}',
                order_by: '={{JSON.parse($parameter.orderBy)}}',
                shard_key: '={{JSON.parse($parameter.shardKey)}}',
            },
            qs: {
                consistency: '={{$parameter.consistency}}',
                timeout: '={{$parameter.timeout}}',
            },
        },
    },
};
exports.scrollPointsFields = [
    (0, Commons_1.collectionNameField)('scrollPoints'),
    {
        displayName: 'Limit',
        name: 'limit',
        default: 50,
        type: 'number',
        typeOptions: {
            minValue: 1,
        },
        description: 'Max number of results to return',
        displayOptions: {
            show: {
                operation: ['scrollPoints'],
            },
        },
    },
    (0, Commons_1.filterField)('scrollPoints'),
    (0, Commons_1.withPayloadField)('scrollPoints'),
    (0, Commons_1.withVectorField)('scrollPoints'),
    {
        displayName: 'Offset',
        name: 'offset',
        description: 'Start ID to read points from',
        default: '',
        type: 'string',
        displayOptions: {
            show: {
                operation: ['scrollPoints'],
            },
        },
    },
    {
        displayName: 'Order By',
        name: 'orderBy',
        description: 'Order the records by a payload field',
        default: 'null',
        type: 'json',
        displayOptions: {
            show: {
                operation: ['scrollPoints'],
            },
        },
    },
    (0, Commons_1.shardKeyField)('scrollPoints'),
    (0, Commons_1.consistencyField)('scrollPoints'),
    (0, Commons_1.timeoutField)('scrollPoints'),
];
