// n8n-nodes-qdrant/dist/nodes/Qdrant/descriptions/SetPayload.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.setPayloadFields = exports.setPayloadOperation = void 0;
const Commons_1 = require("./Commons");
exports.setPayloadOperation = {
    name: 'Set Payload',
    value: 'setPayload',
    action: 'Set Payload',
    description: 'Sets payload values for specified points',
    routing: {
        request: {
            method: 'POST',
            url: '=/collections/{{$parameter.collectionName}}/points/payload',
            body: {
                payload: '={{JSON.parse($parameter.payload)}}',
                points: '={{JSON.parse($parameter.points)}}',
                filter: '={{JSON.parse($parameter.filter)}}',
                shard_key: '={{JSON.parse($parameter.shardKey)}}',
                key: '={{$parameter.key ? $parameter.key : null}}',
            },
            qs: {
                wait: '={{$parameter.wait}}',
                ordering: '={{$parameter.ordering}}',
            },
        },
    },
};
exports.setPayloadFields = [
    (0, Commons_1.collectionNameField)('setPayload'),
    {
        displayName: 'Payload',
        name: 'payload',
        description: 'Payload values to set for the points',
        default: '{}',
        type: 'json',
        required: true,
        displayOptions: {
            show: {
                operation: ['setPayload'],
            },
        },
    },
    (0, Commons_1.pointIdsField)('setPayload'),
    (0, Commons_1.filterField)('setPayload'),
    {
        displayName: 'Key',
        name: 'key',
        description: 'Assigns payload to each point that satisfy this path of property',
        default: '',
        type: 'string',
        displayOptions: {
            show: {
                operation: ['setPayload'],
            },
        },
    },
    (0, Commons_1.shardKeyField)('setPayload'),
    (0, Commons_1.waitField)('setPayload'),
    (0, Commons_1.orderingField)('setPayload'),
];
