// n8n-nodes-qdrant/dist/nodes/Qdrant/descriptions/UpdateVectors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.updateVectorsFields = exports.updateVectorsOperation = void 0;
const Commons_1 = require("./Commons");
exports.updateVectorsOperation = {
    name: 'Update Vectors',
    value: 'updateVectors',
    action: 'Update Vectors',
    description: 'Update specified vectors on points',
    routing: {
        request: {
            method: 'PUT',
            url: '=/collections/{{$parameter.collectionName}}/points/vectors',
            body: {
                points: '={{JSON.parse($parameter.points)}}',
                shard_key: '={{JSON.parse($parameter.shardKey)}}',
            },
            qs: {
                wait: '={{$parameter.wait}}',
                ordering: '={{$parameter.ordering}}',
            },
        },
    },
};
exports.updateVectorsFields = [
    (0, Commons_1.collectionNameField)('updateVectors'),
    {
        displayName: 'Points',
        name: 'points',
        description: 'List of points with their vectors to update. Each point should have ID and vector(s).',
        default: '[\n    {\n        "id": 0,\n        "vector": [\n            0.9,\n            0.1,\n            0.1\n        ]\n    },\n    {\n        "id": 1,\n        "vector": [\n            0.1,\n            0.9,\n            0.1\n        ]\n    }\n]',
        type: 'json',
        required: true,
        displayOptions: {
            show: {
                operation: ['updateVectors'],
            },
        },
    },
    (0, Commons_1.shardKeyField)('updateVectors'),
    (0, Commons_1.waitField)('updateVectors'),
    (0, Commons_1.orderingField)('updateVectors'),
];
