// n8n-nodes-qdrant/dist/nodes/Qdrant/descriptions/UpsertPoints.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.upsertPointsFields = exports.upsertPointsOperation = void 0;
const Commons_1 = require("./Commons");
exports.upsertPointsOperation = {
    name: 'Upsert Points',
    value: 'upsertPoints',
    action: 'Upsert Points',
    description: 'Insert or update points in a collection',
    routing: {
        request: {
            method: 'PUT',
            url: '=/collections/{{$parameter.collectionName}}/points',
            body: {
                points: '={{JSON.parse($parameter.points)}}',
            },
            qs: {
                wait: '={{$parameter.wait}}',
                ordering: '={{$parameter.ordering}}',
            },
        },
    },
};
exports.upsertPointsFields = [
    (0, Commons_1.collectionNameField)('upsertPoints'),
    {
        displayName: 'Points',
        name: 'points',
        description: 'List of points to upsert. Each point should have ID, vector, and optional payload.',
        default: '[\n    {\n        "id": 0,\n        "payload": {\n            "color": "red",\n            "age": 32\n        },\n        "vector": [\n            0.9,\n            0.1,\n            0.1\n        ]\n    },\n    {\n        "id": 1,\n        "payload": {\n            "color": "green",\n            "age": 55\n        },\n        "vector": [\n            0.1,\n            0.9,\n            0.1\n        ]\n    }\n]',
        type: 'json',
        required: true,
        displayOptions: {
            show: {
                operation: ['upsertPoints'],
            },
        },
    },
    (0, Commons_1.waitField)('upsertPoints'),
    (0, Commons_1.orderingField)('upsertPoints'),
];
