// n8n-nodes-qdrant/dist/nodes/Qdrant/descriptions/utils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.listSearch = void 0;
exports.listSearch = {
    async listCollections() {
        try {
            const creds = await this.getCredentials('qdrantRestApi');
            const response = (await this.helpers.httpRequest.call(this, {
                method: 'GET',
                baseURL: creds.url,
                url: 'collections',
                headers: {
                    'api-key': creds.apiKey,
                },
            }));
            if (!response || !response.result || !Array.isArray(response.result.collections)) {
                throw new Error('Invalid response from the Qdrant API');
            }
            const collections = response.result.collections.map((collection) => ({
                name: collection.name,
                value: collection.name,
            }));
            return {
                results: collections,
            };
        }
        catch (error) {
            throw new Error(`Failed to list collections: ${error.message}`);
        }
    },
};
